### PR TITLE
Make user secrets lazy load.

### PR DESCRIPTION
**Pull request by mitchdenny**
Turns user secret loading into a lazy operation that is awaited in each resource provisioning operation.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2998)

### CODE REVIEW

File: src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs [hunk 1]
```diff
@@ -173,9 +182,11 @@ async Task AfterProvisionAsync(IAzureResource resource)
             {
                 var provisioningContext = await provisioningContextLazy.Value.ConfigureAwait(false);
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs:183]: Remove this.

**Review by davidfowl — APPROVED** [review]

**Review by davidfowl — CHANGES_REQUESTED** [review]
Hmm, I just realized this isn't right. I might have been wrong here 😭. We can't save this as it'll overwrite the file.

**Comment by mitchdenny** [comment]
> Hmm, I just realized this isn't right. I might have been wrong here 😭. We can't save this as it'll overwrite the file.

I don't follow. It seems to work as expected.

**Review by davidfowl — APPROVED** [review]

**Pull request merged**